### DISCUSSION TITLE
what does $ and ^ do?

### DISCUSSION THREAD

**Discussion by littleblack111** [hyprwm/Hyprland#7996] [General]
I dont see a difference with or without it. half of the wiki examples are with it and half aren't. shall I use it or is it going to make hyprland more laggy cuz of gonna use more cpu for regex

**Comment by fufexan** [comment]
`^` matches the beginning of the string, `$` matches the end of the string. Should not matter whether they're used or not.

**Comment by littleblack111** [comment]
shall i use it though...

**Reply by MahouShoujoMivutilde** [reply]
```ini
windowrulev2 = bordersize 20, title:^full title 123$
```

will match windows with title

```
full title 123
```

While 

```ini
windowrulev2 = bordersize 20, title:full title 123
```

will match windows with titles

```
full title 123
full title 123 - chromium
foobar full title 123
```

**Comment by littleblack111** [comment]
ohh alright. so ^ and $ makes it more strict. ty!
